### PR TITLE
Null safety for some primitives (#2).

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart [hunk 1]
```diff
@@ -276,13 +270,13 @@ class CustomPointerScrollable extends StatefulWidget {
...
-    final _ScrollableScope widget = context
+    final _ScrollableScope? widget = context
         .getElementForInheritedWidgetOfExactType<_ScrollableScope>()
-        ?.widget as _ScrollableScope;
+        ?.widget as _ScrollableScope?;
     if (widget == null) {
       return false;
     }
-    return widget.position.recommendDeferredLoading(context);
+    return widget.position!.recommendDeferredLoading(context);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart:279]: position should not be nullable. Referencing the flutter framework from which a lot of this class was copied https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/scrollable.dart#L366

File: packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart [hunk 2]
```diff
@@ -332,22 +326,22 @@ class _CustomPointerScrollableState extends State<CustomPointerScrollable>
   /// To control what kind of [ScrollPosition] is created for a [Scrollable],
   /// provide it with custom [ScrollController] that creates the appropriate
   /// [ScrollPosition] in its [ScrollController.createScrollPosition] method.
-  ScrollPosition get position => _position;
-  ScrollPosition _position;
+  ScrollPosition? get position => _position;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart:329]: force unwrap _position and make this non nullable
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/scrollable.dart#L391

File: packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart [hunk 3]
```diff
@@ -681,16 +677,14 @@ class _CustomPointerScrollableState extends State<CustomPointerScrollable>
...
-        super(key: key, child: child);
+    Key? key,
+    required this.scrollable,
+    required this.position,
+    required Widget child,
+  }) : super(key: key, child: child);
 
   final _CustomPointerScrollableState scrollable;
-  final ScrollPosition position;
+  final ScrollPosition? position;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart:687]: make this non nullable

File: packages/devtools_app/lib/src/primitives/enum_utils.dart [hunk 4]
```diff
@@ -29,7 +27,7 @@ class EnumUtils<T> {
   final Map<String, T> _lookupTable = {};
   final Map<T, String> _reverseLookupTable = {};
 
-  T enumEntry(String enumName) => _lookupTable[enumName];
+  T? enumEntry(String? enumName) => _lookupTable[enumName!];
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/primitives/enum_utils.dart:30]: if we are going to always force unwrap `enumName` maybe we shouldn't allow the parameter to be nullable

File: packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/primitives/custom_pointer_scroll_view.dart:329]: tests start failing if I do it
for example, this one: flutter test test/extent_delegate_list_view_test.dart

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested custom_pointer_scroll_view.dart

**Comment by polina-c** [comment]
tested enum_utils

**Comment by polina-c** [comment]
tested extent_delegate_list

**Comment by polina-c** [comment]
tested geometry.dart

**Comment by polina-c** [comment]
tested history_manager.dart

**Pull request merged**